crates/cli/src/native_query/type_solver/simplify.rs

```diff
@@ -212,61 +268,16 @@ fn simplify_constraint_pair(
                 object_type_name: b,
             },
         ) if a == b => match merge_object_type_constraints(context, variable, &a, &b) {
-            Some(merged_name) => Ok(C::Predicate {
+            Some(merged_name) => Ok(Some(C::Predicate {
                 object_type_name: merged_name,
-            }),
-            None => Err((
-                C::Predicate {
-                    object_type_name: a,
-                },
-                C::Predicate {
-                    object_type_name: b,
-                },
-            )),
+            })),
+            None => Ok(None),
         },
 
-        // TODO: We probably want a separate step that swaps ElementOf and FieldOf constraints with
-        // constraint of the targeted structure. We might do a similar thing with
-        // WithFieldOverrides.
```

Comment:
Follow up tickets?

Reply:
I removed the TODO because I don't think we need to worry about this